### PR TITLE
add helmrelease package

### PR DESCRIPTION
**Pull request by xh3b4sd**
We want to retire `e2e-harness` and need to move certain functionality into `e2esetup`. Depends on https://github.com/giantswarm/e2esetup/pull/21 to make the build work.

### CODE REVIEW

File: helmrelease/condition_set.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	v1 "k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type conditionSetConfig struct {
+	K8sClients *k8s.Clients
```
Comment (xh3b4sd) [inline helmrelease/condition_set.go:17]: Our k8s client collection is used here, which also has the ability to provide a rest config.

File: internal/filelogger/filelogger.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+const (
+	folder = "logs"
+)
+
+type Config struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+type FileLogger struct {
```
Comment (xh3b4sd) [inline internal/filelogger/filelogger.go:26]: The file logger magic is also copied from `e2e-harness`.

File: helmrelease/release.go [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+	}
+
+	return nil
+}
+
+// TODO Remove once we are done with migration to EnsureInstalled.
+//
+//	Issue https://github.com/giantswarm/giantswarm/issues/4355.
+//
+func (r *Release) InstallOperator(ctx context.Context, name string, chartInfo ChartInfo, values string, crd *apiextensionsv1beta1.CustomResourceDefinition) error {
```
Comment (kopiczko) [inline helmrelease/release.go]: I'd like to drop this. EnsureInstalled will handle that and we shouldn't install operators with `InstallOperator`.

Comment (xh3b4sd) [reply helmrelease/release.go]: Removed.

**Review by MarcelMue — APPROVED** [review]
Fine with me.

**Review by kopiczko — APPROVED** [review]

**Pull request merged**